tests/lib/config.js

```diff
@@ -1359,6 +1366,108 @@ describe("Config", () => {
                 assertConfigsEqual(actual, expected);
             });
         });
+
+        describe("deprecation warnings", () => {
+            let warning = null;
+
+            function onWarning(w) { // eslint-disable-line require-jsdoc
+                warning = w;
```

Comment:
Should we do something like this instead?

```js
function onWarning(w) {
    if (w.code === 'ESLint') {
        warning = w;
    }
}
```

That way, the test won't break if a warning happens for an unrelated reason (e.g. a deprecation in a dependency).